clang/test/SemaSYCL/intel-restrict.cpp

```diff
@@ -0,0 +1,33 @@
+// RUN: %clang %s -fsyntax-only --sycl -DCHECKDIAG -Xclang -verify
+// RUN: %clang %s -fsyntax-only -Xclang -ast-dump --sycl | FileCheck %s
+
+[[intel::kernel_args_restrict]] // expected-warning{{'kernel_args_restrict' attribute ignored}}
+void func_ignore() {}
```

Comment:
Could you please check that diagnostic emitted when the attribute is applied not to function, i.e. variable, function parameter etc...